src/commands/force/mdapi/beta/retrieve/report.ts

```diff
@@ -0,0 +1,219 @@
+  RetrieveResultFormatter,
+} from '../../../../../formatters/mdapi/retrieveResultFormatter';
+
+interface EnsureFlagOptions {
+  flagName: string;
+  path: string;
+  type: 'dir' | 'file' | 'any';
+  throwOnENOENT?: boolean;
+}
+
+Messages.importMessagesDirectory(__dirname);
+const messages = Messages.loadMessages('@salesforce/plugin-source', 'md.retrieve');
+const spinnerMessages = Messages.loadMessages('@salesforce/plugin-source', 'spinner');
+
+export class Report extends SourceCommand {
+  public static readonly description = messages.getMessage('reportCmd.description');
+  public static readonly examples = messages.getMessage('reportCmd.examples').split(os.EOL);
+  public static readonly requiresUsername = true;
+  public static readonly flagsConfig: FlagsConfig = {
+    // NOTE: this flag is not required because the value is stashed
+    // from the force:mdapi:retrieve command.
+    retrievetargetdir: flags.directory({
+      char: 'r',
+      description: messages.getMessage('flags.retrievetargetdir'),
+      longDescription: messages.getMessage('flagsLong.retrievetargetdir'),
+    }),
+    jobid: flags.id({
+      char: 'i',
+      description: messages.getMessage('flags.jobid'),
+      longDescription: messages.getMessage('flagsLong.jobid'),
+    }),
+    zipfilename: flags.string({
+      char: 'n',
+      description: messages.getMessage('flags.zipfilename'),
+      longDescription: messages.getMessage('flagsLong.zipfilename'),
+    }),
+    unzip: flags.boolean({
+      char: 'z',
+      description: messages.getMessage('flags.unzip'),
+      longDescription: messages.getMessage('flagsLong.unzip'),
+    }),
+    wait: flags.minutes({
+      char: 'w',
+      description: messages.getMessage('flags.wait'),
+      longDescription: messages.getMessage('flagsLong.wait'),
+      default: Duration.minutes(1440), // 24 hours is a reasonable default versus -1 (no timeout)
+    }),
+    verbose: flags.builtin({
+      description: messages.getMessage('flags.verbose'),
+      longDescription: messages.getMessage('flagsLong.verbose'),
+    }),
+  };
+
+  protected retrieveResult: RetrieveResult;
+  protected retrieveStatus: MetadataApiRetrieveStatus;
+  private retrieveTargetDir: string;
+  private zipFileName: string;
+  private unzip: boolean;
+  private wait: Duration;
+  private isAsync: boolean;
+  private mdapiRetrieve: MetadataApiRetrieve;
+
+  public async run(): Promise<RetrieveCommandResult | RetrieveCommandAsyncResult> {
+    await this.report();
+    this.resolveSuccess();
+    return this.formatResult();
+  }
+
+  protected async report(): Promise<void> {
+    // Get stashed retrieve data
+    const mdRetrieveStash = Stash.get<MdRetrieveData>('MDAPI_RETRIEVE');
+
+    const retrieveId = this.resolveRetrieveId(mdRetrieveStash);
+    this.retrieveTargetDir = this.resolveOutputDir(mdRetrieveStash);
+    this.zipFileName = this.getFlag<string>('zipfilename') || mdRetrieveStash.zipfilename || 'unpackaged.zip';
```

Comment:
let's always use `??` (it shouldn't matter for a string type, but it's a good habit)

Reply:
It does matter though.  I want empty string to be treated as false or unset.